spec/unit/environment/with_modules_spec.rb

```diff
@@ -21,90 +21,54 @@
   # Default no additional params
   let(:subject_params) { {} }
 
-  describe "dealing with Puppetfile module conflicts" do
+  describe "dealing with module conflicts" do
     context "with no module conflicts" do
       it "validates when there are no conflicts" do
-        mod = instance_double('R10K::Module::Base', :name => 'nonconflict')
-        expect(subject.puppetfile).to receive(:load)
-        expect(subject.puppetfile).to receive(:modules).and_return [mod]
-        expect { subject.validate_no_module_conflicts }.not_to raise_error
+        mod = instance_double('R10K::Module::Base', name: 'nonconflict', origin: :puppetfile)
+        expect { subject.module_conflicts?(mod) }.not_to raise_error
```

Comment:
Can we check that this returns `false` rather than just not erroring? I think that would be more accurate. This would still pass if it caught conflicts but overrode them, right?